app/controllers/orders_controller.rb

```diff
@@ -75,7 +75,20 @@ def validate_order_request
     end
 
     def skip_recaptcha?
-      (action_name == "create" && params.fetch(:line_items, {}).all? { |product| !Link.find_by(unique_permalink: product["permalink"]).require_captcha? && product["perceived_price_cents"].to_s == "0" }) || valid_wallet_payment?
+      site_key = GlobalConfig.get("RECAPTCHA_MONEY_SITE_KEY")
+      return true if Rails.env.development? && site_key.blank?
```

Comment:
> It would also reduce test flakiness and dependency on the CAPTCHA service.

To reduce the test flakiness due to captcha, we should skip captcha on test env too, right? I guess we can align this condition on both backend and frontend, maybe something like this:

```
(Rails.env.development? || Rails.env.test?) && site_key.blank?
```

What do you think?

Reply:
Thanks for updating it. It would have worked, but would have behaved differently in `staging` when `recaptchaKey` is not set. Similar to `production`, we want `staging` to require `recaptchaKey` and throw errors if it's not present.